### ISSUE TITLE
Your 1.0.6 release broke backwards compatiblity with node 0.10

### ISSUE THREAD

**Issue by thomaseizinger** [nfroidure/streamfilter#4]
With the latest release of your library it is no longer compatible with old node versions.
This might be intended since you write in your issue template that you do not maintain versions below 6.9.5.

This is fine, however, you should have released a new major version for this because now your package is incompatible with old node versions. I don't know the exact change that broke it but due to node using version ranges, we suddenly had a breaking build of our mobile website without changing anything.

So, you might consider raising that version to a major one and somehow drop the old one (if that is possible on npm, I don't know).

**Comment by nfroidure** [comment]
You are perfectly right. Nevermind those changes would break 0.10, neither people were continuing using such an old version of Node since its maintenance is not guaranteed.

Try to pin the version to the prior one as a short term fix.

If you wish, I can give you access to the repo/node module to take in charge the Node 0.10 long term support. In this case, I think that reverting, patching and then publishing the patch will be enough.

**Comment by vcfvct** [comment]
The `gulp-filter` has dependency on `"streamfilter": "^1.0.5"`, it breaks our build which stunk in node 0.12....

**Comment by thomaseizinger** [comment]
@vcfvct Same thing happened to our build which forced us to lock the version of streamfilter to 1.0.5.

I don't expect actual support for the code for such an old version, it is just that due to the version range, builds start to break "randomly".
It would be really nice if you would just remove version 1.0.6 from npm and republish it as 2.0.0. As 1.0.6 is only 2 days old, I don't expect anyone to have manually installed it (e.g. actually depending on that version instead of accidentally getting it through a version range).
I think that at least some harm could be prevented by doing so, since there are now already two projects affected with that. (And it took one of our team members a few hours to track down why the build was broken.)

Thanks in advance @nfroidure !

In summary: The biggest problem here is that `streamfilter` is a __transitive__ dependency in a lot of projects that use gulp and builds are breaking because of a breaking change in 1.0.6 that is incorrectly labelled as a patch-release (which thereby causes npm to assume it is safe to pick the next version) instead of a major release as it should be for breaking changes.

**Comment by vcfvct** [comment]
@thomaseizinger , right, for now, I have to fork a old version of `gulp-filter` and fix the dependency on `1.0.5` as a workaround.

**Comment by thomaseizinger** [comment]
Not necessarily. If you use shrinkwrap then you can override the version and specify the one you need. This is what we did.

**Comment by vcfvct** [comment]
@thomaseizinger , we did not use shrinkwrap in that repo.  We are in a code base transition period and will use webpack in new codebase instead of gulp for building. 
Thanks for the tips.

**Comment by nfroidure** [comment]
@thomaseizinger can't unpublish on NPM since `leftpad` I guess ;). Anyway I find some time to do it.

Make me a favor though, stop using such old NodeJS versions.

**Comment by thomaseizinger** [comment]
If it would be on me, I would instantly upgrade to latest node. Unfortunately it is not a private project but work and you know how that is. There is never time for maintenance like that :)

Npm says you can contact them for unpublishing of packages that are older than 24h. You might try that :)
